packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -816,8 +820,36 @@ export const Table = <
     }
   });
 
+  const virtualizer = useVirtualScroll({
+    count: page.length,
+    getScrollElement: () => tableElement ?? null,
+    estimateSize: () => {
+      if (density === 'condensed') {
+        return 50;
+      } else if (density === 'extra-condensed') {
+        return 38;
+      }
+      return 62;
+    }, //Set to the height of the table row based on the value of the density prop.
+    scrollToIndex,
+  });
+
+  const outerProps = {
+    style: {
+      minBlockSize: virtualizer.getTotalSize(),
+      minInlineSize: '100%',
+      willChange: 'transform',
```

Comment:
I believe the `willChange: "transform"` can be removed from all three components. We are no longer using a `transform` on the wrapping div

Reply:
Also I noticed in the tanstack-virtual docs that they add [`contain: strict`](https://developer.mozilla.org/en-US/docs/Web/CSS/contain) to the outer div. It probably helps with performance so we should do the same

Reply:
It might produce a bigger performance gain on the outer div? Lets add an after-PR TODO linking to this thread to investigate more.

For now, we can use `contain: strict` on the inner div in all three components.

Also, I think the variable names are confusing. You should rename them all to be something generic like `virtualWrapperProps`, since we only have a single div, not two. Alternatively, you could even inline the props directly instead of using an intermediate variable.

Reply:
I inlined the props directly to the `<div>` in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/a95f14d684f51bcb33977b13457be9dc14782561). Also added the after PR TODO and moved the `contain: strict` style to the inner `<div>` on the `Tree` PR in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/a3a119fde05d570ad86aeb3c515186d244615398).

Reply:
Can you use the same approach everywhere? If you're inlining it in `Table`, then `ComboBox` and `Tree` should also inline it.

Also I noticed a mistake: the `style` prop should not be merged on L1166. That `style` prop is only meant for the root `Table` element (L950).

Reply:
Removed the spread `style` prop in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/adbc5c3b76a0b00ca5aa3255f0f1416e2ae58fe7). Also inlined the styles for the `Tree` component in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/4d91fd1a8ea51f87dc7c27dbb0871bf205e530b0). The `ComboBox` was already inlining this [here](https://github.com/iTwin/iTwinUI/blob/74211fd2247e3a49e0269557e676edd11fe99e2a/packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx#L97), so that PR hasn't changed.